pandas/core/generic.py

```diff
@@ -6469,15 +6469,45 @@ def fillna(
         else:
             return result.__finalize__(self, method="fillna")
 
+    @overload
+    def ffill(
+        self: FrameOrSeries,
+        axis: None | str | int = ...,
+        inplace: Literal[False] = ...,
+        limit: None | int = ...,
+        downcast: None | str | dict = ...,
+    ) -> FrameOrSeries:
+        ...
+
+    @overload
+    def ffill(
+        self: FrameOrSeries,
+        axis: None | str | int = ...,
+        inplace: Literal[True] = ...,
```

Comment:
this one doesn't need `=...` - else, if you call `.ffill()`, mypy won't know which overload to match

Reply:
there also needs to be an extra overload for when `axis` isn't passed:
```python
    @overload
    def ffill(
        self: FrameOrSeries,
        *,
        inplace: Literal[True],

Reply:
Done